include/votca/csg/beadstructure.h

```diff
@@ -76,27 +75,28 @@ class BeadStructure {
    *
    * The same bead cannot be added twice.
    **/
-  void AddBead(T *bead);
+  void AddBead(T &bead);
 
   /**
    * \brief Get the bead with the specified id
    **/
-  T *getBead(Index id);
+  T &getBead(const Index &id);
+  const T &getBead(const Index &id) const;
 
   /**
    * \brief Create a connection between two beads in the structure
    *
    * A bead cannot be connected to itself. It also may not be connected to a
    * bead that has not yet been added to the structure.
    **/
-  void ConnectBeads(Index bead1_id, Index bead2_id);
+  void ConnectBeads(const Index &bead1_id, const Index &bead2_id);
 
   /**
    * \brief Return a vector of all the beads neighboring the index
    **/
   std::vector<T *> getNeighBeads(Index index);
 
-  TOOLS::Graph getGraph();
+  tools::Graph getGraph();
```

Comment:
can it be const?

Reply:
It is a copy there is no need to make const, and the copy is needs to be mutable so no, but good question.

Reply:
I meant "Graph getGraph() const` not `const Graph getGraph()` sorry for the confusion

Reply:
No, unfortunately, it cannot be at this point. The graph is only generated internally when it is needed. When the getGraph method is called it checks to see if the graph is up to date, if it is it does not generate anything, but if it isn't the graph is generated. I could auto update the graph every time a bead is added but this would be expensive, particularly if the graph is never actually needed.

Reply:
ahh okay